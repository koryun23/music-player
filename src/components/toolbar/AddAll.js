import React, { useState } from "react";
import "../../css/toolbar/Toolbar.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAdd, faCaretDown } from "@fortawesome/free-solid-svg-icons";

export default function AddAll(props) {

    const [isDropdownOpen, setIsDropdownOpen] = useState(false);

    const clickDropdown = () => {
        setIsDropdownOpen(!isDropdownOpen);
    }
    return (
        <div className="add-all-button dropdown dropdown-div" onClick={clickDropdown}>
            <button className="button-description btn btn-light dropdown-toggle"
                    type="button"
                    id="dropdownMenuButton"
                    data-toggle="dropdown"
                    aria-haspopup="true"
                    aria-expanded="false">
                <FontAwesomeIcon icon={faAdd} size="lg" className="add-icon"/>
                <b>Add All</b>
            </button>
            <div className={isDropdownOpen ? "dropdown-menu show" : "dropdown-menu"}>
                <a href="#" className="dropdown-item">Option 1</a>
                <a href="#" className="dropdown-item">Option 1</a>
                <a href="#" className="dropdown-item">Option 1</a>

            </div>
        </div>
    );
}