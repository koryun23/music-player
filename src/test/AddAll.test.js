import renderer from "react-test-renderer";
import AddAll from "../components/toolbar/AddAll";

it('add all renders correctly', () => {
    const tree = renderer.create(<AddAll />).toJSON();
    expect(tree).toMatchSnapshot();
})

it('add all dropdown works correctly', () => {
    const component = renderer.create(<AddAll />);
    let tree = component.toJSON();
    renderer.act(() => {
        tree.props.onClick();
    })
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    renderer.act(() => {
        tree.props.onClick();
    })
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})