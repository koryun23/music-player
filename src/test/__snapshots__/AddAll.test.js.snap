// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add all dropdown works correctly 1`] = `
<div
  className="add-all-button dropdown dropdown-div"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-lg add-icon"
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Add All
    </b>
  </button>
  <div
    className="dropdown-menu show"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Sequentially
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Shuffled
    </a>
  </div>
</div>
`;

exports[`add all dropdown works correctly 2`] = `
<div
  className="add-all-button dropdown dropdown-div"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-lg add-icon"
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Add All
    </b>
  </button>
  <div
    className="dropdown-menu"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Sequentially
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Shuffled
    </a>
  </div>
</div>
`;

exports[`add all renders correctly 1`] = `
<div
  className="add-all-button dropdown dropdown-div"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-lg add-icon"
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Add All
    </b>
  </button>
  <div
    className="dropdown-menu"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Sequentially
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Shuffled
    </a>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="add-all-button dropdown dropdown-div"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-lg add-icon"
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Add All
    </b>
  </button>
  <div
    className="dropdown-menu"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Sequentially
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Add All Shuffled
    </a>
  </div>
</div>
`;
